src/Middleware/HttpLogging/src/BufferingStream.cs

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal class BufferingStream : Stream, IBufferWriter<byte>
```

Comment:
```suggestion
    internal sealed class BufferingStream : Stream, IBufferWriter<byte>
```

Reply:
This is a base class to RequestBufferingStream and ResponseBufferingStream. Purposefully not sealed.